From 898a061d2508b7863c4f19929db5d9d7fe08cb2c Mon Sep 17 00:00:00 2001
From: Gabriel Ravier <gabravier@gmail.com>
Date: Thu, 4 Jul 2024 12:30:25 +0200
Subject: [PATCH] Add UTIME_NOW, UTIME_OMIT and fix declaration of utimensat
To: dietlibc@fefe.de

UTIME_NOW and UTIME_OMIT are required by POSIX.
The prototype for utimensat as it was before now is non-compliant, as the third parameter must be a const pointer.
---
 include/sys/stat.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/sys/stat.h b/include/sys/stat.h
index 33ae3374..e338ae44 100644
--- a/include/sys/stat.h
+++ b/include/sys/stat.h
@@ -723,6 +723,9 @@ extern int mkfifo (const char *__path, mode_t __mode) __THROW;
 #define S_IWRITE S_IWUSR
 #define S_IEXEC S_IXUSR
 
+#define UTIME_NOW ((1ul << 30) - 1ul)
+#define UTIME_OMIT ((1ul << 30) - 2ul)
+
 /* also include fcntl.h for the AT_* constants */
 
 int fchmodat(int dirfd, const char *pathname, mode_t mode, int flags) __THROW;
@@ -730,7 +733,7 @@ int fstatat(int dirfd, const char *pathname, struct stat *buf, int flags) __THRO
 int mkdirat(int dirfd, const char *pathname, mode_t mode) __THROW;
 int mknodat(int dirfd, const char *pathname, mode_t mode, dev_t dev) __THROW;
 int mkfifoat(int dirfd, const char *pathname, mode_t mode) __THROW;
-int utimensat(int dirfd, const char *pathname, struct timespec* t,int flags) __THROW;
+int utimensat(int dirfd, const char *pathname, const struct timespec* t,int flags) __THROW;
 
 __END_DECLS
 
-- 
2.45.2

